truffle/com.oracle.truffle.api/src/com/oracle/truffle/api/nodes/LoopNode.java

```diff
@@ -24,15 +24,96 @@
  */
 package com.oracle.truffle.api.nodes;
 
+import com.oracle.truffle.api.TruffleRuntime;
 import com.oracle.truffle.api.frame.VirtualFrame;
 
 /**
- * Experimental API: may change significantly without notice.
+ * A loop node calls {@link RepeatingNode#executeRepeating(VirtualFrame) repeating nodes} as long as
+ * it returns <code>true</code>. Using the loop node in a guest language implementation allows the
+ * Truffle runtime to optimize loops in a better way. For example a Truffle runtime implementation
+ * might decide to optimize loop already during its first execution (also called on stack
+ * replacement OSR). Loop nodes are not intended to be implemented by Truffle runtime
+ * implementations and not by guest language implementations.
+ *
+ * Full usage example for guest language while node:
+ *
+ * <pre>
+ * public class WhileNode extends GuestLanguageNode {
+ *
+ *     &#064;Child private {@link LoopNode} loop;
+ *
+ *     public WhileNode(GuestLanguageNode conditionNode, GuestLanguageNode bodyNode) {
+ *         loop = Truffle.getRuntime().createLoopNode(new WhileRepeatingNode(conditionNode, bodyNode));
+ *     }
+ *
+ *     &#064;Override
+ *     public Object execute({@link VirtualFrame} frame) {
+ *         loop.executeLoop(frame);
+ *         return null;
+ *     }
+ *
+ *     private static class WhileRepeatingNode extends {@link Node} implements {@link RepeatingNode} {
+ *
+ *         &#064;Child private GuestLanguageNode conditionNode;
+ *         &#064;Child private GuestLanguageNode bodyNode;
+ *
+ *         public WhileRepeatingNode(GuestLanguageNode conditionNode, GuestLanguageNode bodyNode) {
+ *             this.conditionNode = conditionNode;
+ *             this.bodyNode = bodyNode;
+ *         }
+ *
+ *         public boolean executeRepeating({@link VirtualFrame} frame) {
+ *             if ((boolean) conditionNode.execute(frame)) {
+ *                 try {
+ *                     bodyNode.execute(frame);
+ *                 } catch (ContinueException ex) {
+ *                     // the body might throw a continue control-flow exception
+ *                     // continue loop invocation
+ *                 } catch (BreakException ex) {
+ *                     // the body might throw a break control-flow exception
+ *                     // break loop invocation by returning false
+ *                     return false;
+ *                 }
+ *                 return true;
+ *             } else {
+ *                 return false;
+ *             }
+ *         }
+ *     }
+ *
+ * }
+ *
+ * // substitute with a guest language node type
+ * public abstract class GuestLanguageNode extends Node {
+ *
+ *     public abstract Object execute(VirtualFrame frame);
+ *
+ * }
+ * // thrown by guest language continue statements
+ * public final class ContinueException extends ControlFlowException {}
+ * // thrown by guest language break statements
+ * public final class BreakException extends ControlFlowException {}
```

Comment:
One more comment. I believe that 
```
public final class BreakException extends Something
```
should actually be written as
```html
<b>public final class</b> BreakException <b>extends</b> Something
```
to make the code somewhat readable in Javadoc. Again, very hard to do it manually and apply it to all keywords.